frontend/OBSApp.cpp

```diff
@@ -1748,31 +1818,66 @@ void OBSApp::SigIntSignalHandler(int s)
 	char a = 1;
 	send(sigintFd[0], &a, sizeof(a), 0);
 }
+
+void OBSApp::SigTermSignalHandler(int s)
+{
+	blog(LOG_INFO, "[OBSApp] SigTermSignalHandler");
+
+	UNUSED_PARAMETER(s);
+
+	char a = 1;
+	send(sigtermFd[0], &a, sizeof(a), 0);
+}
 #endif
 
 void OBSApp::ProcessSigInt(void)
 {
 	/* This looks weird, but we can't ifdef a Qt slot function so
 	 * the SIGINT handler simply does nothing on Windows. */
 #ifndef _WIN32
+	blog(LOG_INFO, "[OBSApp] ProcessSigInt");
+
 	char tmp;
 	recv(sigintFd[1], &tmp, sizeof(tmp), 0);
 
 	OBSBasic *main = OBSBasic::Get();
-	if (main)
+	if (main) {
+		main->saveAll();
 		main->close();
+	}
+#endif
+}
+
+void OBSApp::ProcessSigTerm(void)
+{
+#ifndef _WIN32
+	blog(LOG_INFO, "[OBSApp] ProcessSigTerm");
+
+	char tmp;
+	recv(sigtermFd[1], &tmp, sizeof(tmp), 0);
+
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		main->saveAll();
+	}
+
+	quit();
 #endif
 }
 
-#ifdef _WIN32
 void OBSApp::commitData(QSessionManager &manager)
 {
-	if (auto main = App()->GetMainWindow()) {
-		QMetaObject::invokeMethod(main, "close", Qt::QueuedConnection);
-		manager.cancel();
+	OBSBasic *main = OBSBasic::Get();
+	if (main) {
+		blog(LOG_INFO, "[OBSApp] commitData");
+		main->saveAll();
+
+		if (manager.allowsInteraction() && main->shouldPromptForClose()) {
+			blog(LOG_INFO, "[OBSApp] SessionManager::cancel()");
```

Comment:
If we do defensive verbose logging for remote debugging, the messages should at least be more verbose and have some user value. Otherwise this should be `DEBUG` level.